Library/Homebrew/test/spec_helper.rb

```diff
@@ -10,10 +10,11 @@
   ]
   SimpleCov.formatters = SimpleCov::Formatter::MultiFormatter.new(formatters)
 
+  # Needed for outputting coverage reporting only once for parallel_tests
   if RUBY_PLATFORM[/darwin/] && ENV["TEST_ENV_NUMBER"]
     SimpleCov.at_exit do
       result = SimpleCov.result
-      result.format! if ParallelTests.number_of_running_processes <= 1
+      result.format! if ParallelTests.last_process?
```

Comment:
I'm not very familiar with the `parallel_tests` gem but according to the docs it looks like this `.last_process?` method indicates that it's the last started process not the last finished process. I assume we want to test for the last finished process so that we format the coverage result for all test processes. If that's the case, we should keep the old logic.

- https://github.com/grosser/parallel_tests?tab=readme-ov-file#running-setup-or-teardown-once

Reply:
Yeah this would need `ParallelTests.wait_for_other_processes_to_finish` for this to work.

Reply:
The bit that's weird here is the `RUBY_PLATFORM[/darwin/]`. Why not do this for all OS?

Reply:
> I'm not very familiar with the `parallel_tests` gem but according to the docs

@apainintheneck I am very familiar with it. I wrote this code (3c270b3a9a9b77db9bf511ce40891f4311584135). I also reread all the relevant SimpleCov code in making this change.

> `.last_process?` method indicates that it's the last started process not the last finished process

Yes but: this doesn't matter. We don't need to run this on the last running process, just any single process.

> Yeah this would need `ParallelTests.wait_for_other_processes_to_finish` for this to work which internally just does `number_of_running_processes <= 1` anyway but with the downside of sleep calls.

@Bo98 did you test this locally? I did and it seemed to work as expected. Similarly, I don't understand why the coverage metrics on this PR would work if this were the case.

I'm going to just leave this as-is, I'm not sure it's worth anyone's time to further discuss this cleanup. I don't think the review process here has been optimal.

> The bit that's weird here is the `RUBY_PLATFORM[/darwin/]`. Why not do this for all OS? I suspect that part of the check came from a time when coverage was macOS only.

@Bo98 Feel free to open a PR for this if desired.

Reply:
> I did and it seemed to work as expected. Similarly, I don't understand why the coverage metrics on this PR would work if this were the case.

Looks like that's because SimpleCov already has ParallelTests support and has the `ParallelTests.wait_for_other_processes_to_finish` call I suggested adding: https://github.com/simplecov-ruby/simplecov/blob/main/lib/simplecov.rb#L110.

I'll clarify the comment to specify this more directly to avoid future confusion here - I thought it meant that SimpleCov didn't support parallel tests at all. Probably worth upstreaming this diff at some point.

Reply:
@Bo98 Thanks for getting this over the line!